flytekit/image_spec/image_spec.py

```diff
@@ -143,6 +144,9 @@ def exist(self) -> Optional[bool]:
             if e.response.status_code == 404:
                 return False
 
+            if re.match(f"unknown: repository .*{self.name} not found", e.explanation):
```

Comment:
don't we want to show the error message as well?

Reply:
I just added a print out with the explanation. Let me know if looks okay.